jsonnet/components/prometheus.libsonnet

```diff
@@ -316,6 +341,7 @@ function(params)
           'kube-rbac-proxy',
           'metrics-client-certs',
         ],
+        externalURL: 'https://prometheus-k8s.openshift-monitoring.svc:9091',
```

Comment:
hmm I missed this detail :thinking:...
Ideally I would expect that we link back to the OCP console but the generated URL is hardcoded to `<externalURL>/graph?g0.expr=up&g0.tab=1` which wouldn't work for the console. Maybe @kyoto has some ideas?

Reply:
Not sure I follow...link back from where? Afaiu this is what the Prometheus UI uses internally. With this PR it should only be accessible through the Prometheus Service, so setting this as the external URL should suffice.

Reply:
Sorry for the missing details :) 
When Prometheus sends alerts to Alertmanager, the payload includes a `generatorURL` field that is a back-link to the Prometheus UI (see https://prometheus.io/docs/alerting/latest/clients/). The URL is constructed by concatenating `--web.external-url` and `/graph?g0.expr=<alert expression>&g0.tab=1` (see [here](https://github.com/prometheus/prometheus/blob/a5d57c926903677e0e16c29bce46de69dfca153f/cmd/prometheus/main.go#L1230) and [here](https://github.com/prometheus/prometheus/blob/a5d57c926903677e0e16c29bce46de69dfca153f/util/strutil/strconv.go#L26-L29)).
My thinking is that if the generator URL would link to the console route and the console redirects to the "Observe > Metrics" page... The other option would be to discuss upstream if it would be possible to have a full customization of the generator URL (not sure if it makes sense in general).

Reply:
Adding a Console redirect sounds like a reasonable workaround to me.

Can we have the generator URL go directly to `<Console URL>/monitoring` in the Console? Then, if I understand correctly, the constructed URL would be `<Console URL>/monitoring/graph?g0.expr=<alert expression>&g0.tab=1`, which Console would then redirect to `<Console URL>/monitoring/query-browser?query0=<alert expression>`?

Reply:
Looks good @kyoto, lets try it.

Reply:
PR for handling the Console redirect: https://github.com/openshift/console/pull/10963